embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/capture/crash/CrashDataSourceImplTest.kt

```diff
@@ -0,0 +1,181 @@
+package io.embrace.android.embracesdk.capture.crash
+
+import io.embrace.android.embracesdk.FakeNdkService
+import io.embrace.android.embracesdk.assertJsonMatchesGoldenFile
+import io.embrace.android.embracesdk.deserializeJsonFromResource
+import io.embrace.android.embracesdk.fakes.FakeAnrService
+import io.embrace.android.embracesdk.fakes.FakeConfigService
+import io.embrace.android.embracesdk.fakes.FakeLogOrchestrator
+import io.embrace.android.embracesdk.fakes.FakeLogWriter
+import io.embrace.android.embracesdk.fakes.FakePreferenceService
+import io.embrace.android.embracesdk.fakes.FakeSessionOrchestrator
+import io.embrace.android.embracesdk.fakes.fakeEmbraceSessionProperties
+import io.embrace.android.embracesdk.internal.crash.CrashFileMarker
+import io.embrace.android.embracesdk.internal.serialization.EmbraceSerializer
+import io.embrace.android.embracesdk.logging.InternalEmbraceLogger
+import io.embrace.android.embracesdk.payload.Crash
+import io.embrace.android.embracesdk.payload.JsException
+import io.embrace.android.embracesdk.payload.LegacyExceptionInfo
+import io.embrace.android.embracesdk.payload.ThreadInfo
+import io.embrace.android.embracesdk.session.properties.EmbraceSessionProperties
+import io.embrace.android.embracesdk.utils.at
+import io.mockk.mockk
+import io.mockk.unmockkAll
+import io.mockk.verify
+import org.junit.After
+import org.junit.Assert
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+
+internal class CrashDataSourceImplTest {
+
+    private lateinit var crashDataSource: CrashDataSourceImpl
+    private lateinit var logOrchestrator: FakeLogOrchestrator
+    private lateinit var sessionOrchestrator: FakeSessionOrchestrator
+    private lateinit var sessionProperties: EmbraceSessionProperties
+    private lateinit var anrService: FakeAnrService
+    private lateinit var ndkService: FakeNdkService
+    private lateinit var preferencesService: FakePreferenceService
+    private lateinit var crashMarker: CrashFileMarker
+    private lateinit var configService: FakeConfigService
+    private lateinit var serializer: EmbraceSerializer
+    private lateinit var logWriter: FakeLogWriter
+    private lateinit var logger: InternalEmbraceLogger
+    private lateinit var localJsException: JsException
+    private lateinit var testException: Exception
+    @Before
+    fun setUp() {
+        logOrchestrator = FakeLogOrchestrator()
+        sessionOrchestrator = FakeSessionOrchestrator()
+        sessionProperties = fakeEmbraceSessionProperties()
+        anrService = FakeAnrService()
+        ndkService = FakeNdkService()
+        preferencesService = FakePreferenceService()
+        crashMarker = mockk(relaxUnitFun = true)
```

Comment:
Don't need this now, but we should add a Fake for this

Reply:
Added a FakeCrashFileMarker